system/updated/casync/common.py

```diff
@@ -0,0 +1,61 @@
+import json
+import pathlib
+import subprocess
+
+from openpilot.system.version import get_release_notes, get_version
+
+
+CASYNC_ARGS = ["--with=symlinks", "--with=permissions"]
+CHANNEL_MANIFEST_FILE = "channel.json" # file that contains details of the current release
+CASYNC_FILES = [CHANNEL_MANIFEST_FILE, ".caexclude"]
+
+
+def run(cmd):
+  return subprocess.check_output(cmd)
+
+
+def get_exclude_set(path) -> set[str]:
+  exclude_set = set(CASYNC_FILES)
+
+  for file in path.rglob("*"):
+    if file.is_file() or file.is_symlink():
+
+      while file.resolve() != path.resolve():
+        exclude_set.add(str(file.relative_to(path)))
+
+        file = file.parent
+
+  return exclude_set
+
+
+def create_caexclude_file(path: pathlib.Path):
+  with open(path / ".caexclude", "w") as f:
+    # exclude everything except the paths already in the release
+    f.write("*\n")
+    f.write(".*\n")
+
+    for file in sorted(get_exclude_set(path)):
+      f.write(f"!{file}\n")
+
+
+def create_manifest(channel, version, release_notes):
```

Comment:
we already use manifest to refer to the file that describes the state of a version https://github.com/commaai/openpilot/blob/master/system/hardware/tici/agnos.json. this is more a version metadata info file